kolibri/plugins/coach/assets/src/views/reports/report-subheading.vue

```diff
@@ -1,38 +1,56 @@
 <template>
-
-  <div v-if="showRecentOnly">
-    <sub v-if="standardDataTable.length">{{ $tr('subHeading', { threshold }) }}</sub>
-    <sub v-else>{{ $tr('noRecentProgress', { threshold }) }}</sub>
+  <div>
+    <sub v-if="pageName == recent && !standardDataTable.length">{{ $tr('noRecentProgress', { threshold }) }}</sub>
+    <sub v-if="pageName == exam && !exams.length ">{{ $tr('noExams') }}</sub>
+    <sub v-if="pageName == general && standardDataTable.length">{{ $tr('subHeading' , { threshold }) }}</sub>
+    <sub v-if="pageName == learners && !standardDataTable.length">{{ $tr('noLearners') }}</sub>
+    <sub v-if="pageName == groups">{{ $tr('noGroups') }}</sub>
+    <!-- <sub v-if="pageName == groups && !group.users.length">{{ $tr('noGroups') }}</sub> TOASK how to obtain the prop  -->
   </div>
-
 </template>
 
 
 <script>
 
   import * as ReportConstants from '../../reportConstants';
   import * as reportGetters from '../../state/getters/reports';
+  // import {pageNames} from '../../constants';
 
   export default {
     name: 'coachReportSubheading',
     $trs: {
+      noRecentProgress: 'No activity in past {threshold} days', //missing translation too ad manually to /kolibri/locale/es_ES/LC_FRONTEND_MESSAGES/coach_module-messages.json ?
+      noExams: 'You do not have any exams. Start by creating a new exam above',
+      noLearners: 'You do not have any learners registered yet',
+      noGroups: 'You do not have any groups created yet',
       subHeading: 'Only showing activity in past {threshold} days',
-      noRecentProgress: 'No activity in past {threshold} days',
     },
+    props: ['group'],
+    data: () => ({
+      // recent: pageNames.RECENT_CHANNELS, TOASK how to include the constants /src/constants.js
+      recent: 'RECENT_CHANNELS',
```

Comment:
These should all be coming from the imported `PAGE_NAMES` constant that you commented out above.

Reply:
yeah the problem it's that when I do:
`import {pageNames} from '../../constants';`
and then on the data funcion:
`recent: pageNames.RECENT_CHANNELS`

it shows the following error
![screen shot 2017-11-22 at 8 31 04 pm](https://user-images.githubusercontent.com/13241119/33156087-4263760c-cfc4-11e7-9372-d7fd54a380f1.png)

Reply:
I believe because it is called `PageNames` not `pageNames`.